# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/EncoderTemplate.h
# Commit: da42eb3f9664
# Full Hash: da42eb3f9664fc2d5dc77db2dc73e45b025c9a31
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 1
# Description:
#   Bug 1749047 - Convert from MediaRawData to EncodedVideoChunk one by one. r=chunmin
#   
#   This simplifies the rest of the code, in remaining patches.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195713
# ==============================================================================

diff -r e59a8dc5add5 -r da42eb3f9664 dom/media/webcodecs/EncoderTemplate.h
--- a/dom/media/webcodecs/EncoderTemplate.h	Fri Dec 22 21:39:09 2023 +0000
+++ b/dom/media/webcodecs/EncoderTemplate.h	Fri Dec 22 21:39:10 2023 +0000
@@ -169,11 +169,12 @@
 
   /* Type conversion functions for the Encoder implementation */
  protected:
-  virtual nsTArray<RefPtr<OutputType>> EncodedDataToOutputType(
+  virtual RefPtr<OutputType> EncodedDataToOutputType(
       nsIGlobalObject* aGlobalObject,
-      nsTArray<RefPtr<MediaRawData>>&& aData) = 0;
+      RefPtr<MediaRawData>& aData) = 0;
   virtual OutputConfigType EncoderConfigToDecoderConfig(
       nsIGlobalObject* aGlobalObject,
+      const RefPtr<MediaRawData>& aData,
       const ConfigTypeInternal& aOutputConfig) const = 0;
   /* Internal member variables and functions */
  protected: